### PR TITLE
feat(store): improve signature types of select method in store

### PR DESCRIPTION
**Pull request by e-cloud**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

the signature of select method doesn't have any type safe assurance.

## What is the new behavior?
better type constraints

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/store/tests/devMode.spec.ts [hunk 1]
```diff
@@ -352,7 +352,7 @@ describe('Development Mode', () => {
 
       const store: Store = TestBed.get(Store);
 
-      const state = store.selectSnapshot<StateModel>(appState => appState.counter);
+      const state = store.selectSnapshot<StateModel, any>(appState => appState.counter);
```
Comment (splincode) [inline packages/store/tests/devMode.spec.ts]: it's breaking change?

File: packages/store/src/store.ts [hunk 2]
```diff
@@ -26,8 +27,9 @@ export class Store {
   /**
    * Selects a slice of data from the store.
```
Comment (splincode) [inline packages/store/src/store.ts]: Could you write a description of generics everywhere as JSDOC?

```ts
/**
 * Selects a slice of data from the store.
 * @template R - what is it?
 * @template K - what is it?
 * @template T - what is it?
*/
```

**Comment by markwhitfeld** [comment]
@e-cloud Are you going to be moving forward with this PR?

**Comment by splincode** [comment]
any news? we could continue later

**Pull request closed without merging**